jsonschema-validation.xml

```diff
@@ -536,84 +530,65 @@
+                    Format-Assertion vocabulary, is:
+                    &lt;https://json-schema.org/draft/2020-11/vocab/format-assertion&gt;. The current
+                    URI for the corresponding meta-schema is:
+                    <eref target="https://json-schema.org/draft/2020-11/meta/format-assertion"/>.
+                    Implementing support for the Format-Assertion vocabulary is OPTIONAL.
                 </t>
                 <t>
-                    The current URI for the corresponding meta-schema is:
-                    <eref target="https://json-schema.org/draft/2020-11/meta/format"/>.
+                    Specifying both the Format-Annotation and the Format-Assertion vocabularies is functionally
+                    equivalent to specifying only the Format-Assertion vocabulary since its requirements
+                    are a superset of the Format-Annotation vocabulary.
                 </t>
-
             </section>
 
             <section title="Implementation Requirements">
                 <t>
-                    The "format" keyword functions as an annotation, and optionally as an assertion.
-                    <cref>This is due to the keyword's history, and is not in line with current
-                    keyword design principles.</cref>  In order to manage this ambiguity, the
-                    "format" keyword is defined in its own separate vocabulary, as noted above.
-                    The true or false value of the vocabulary declaration governs the implementation
-                    requirements necessary to process a schema that uses "format", and the
-                    behaviors on which schema authors can rely.
+                    The "format" keyword functions as defined by the vocabulary which is referenced.
                 </t>
 
-                <section title="As an annotation">
+                <section title="Format-Annotation Vocabulary">
                     <t>
                         The value of format MUST be collected as an annotation, if the implementation
-                        supports annotation collection.  This enables application-level validation when
+                        supports annotation collection. This enables application-level validation when
                         schema validation is unavailable or inadequate.
                     </t>
                     <t>
-                        This requirement is not affected by the boolean value of the vocabulary
-                        declaration, nor by the configuration of "format"'s assertion
-                        behavior described in the next section.
+                        Implementations MAY still treat "format" as an assertion in addition to an
+                        annotation and attempt to validate the value's conformance to the specified
+                        semantics. The implementation MUST provide options to enable and disable such
```

Comment:
Why MUST they, if it can also be done via $vocabulary? I would soften this to a SHOULD. Ideally, the fewer knobs needed in an implementation outside of specifying a `$schema`, the better.

Reply:
This is solely for when the annotation vocabulary is used.  _If_ an implementation supports assertion (even partial assertion) they MAY choose to treat the keyword as such.  If they do, they MUST allow this feature to be enabled and disabled.

The assertion vocabulary doesn't have this allowance, instead specifying that implementations MUST implement assertion fully in order to support the vocabulary.

Reply:
I'm still confused. It sounds like a schema (via the metaschema) can specify to use the format-annotation vocabulary, but it might still get assertion behaviour anyway (with implementations that supported validations)? That would force the setting of an out-of-band configuration "no, really, when I say annotations I mean only annotations" and there would be no other way to not get validation behaviour as well.  If I'm correct so far, then the configuration value MUST default to false so validation behaviour doesn't happen accidentally.

Reply:
The schema can say, "You're getting annotations."

But this allows the client/application to say, "I know I'm supposed to get annotations, but this library I'm using supports asserting formats, and that'd save me some work, so I'd like to use that."  It gives the choice to the client.

It also the opens the door for implementations to provide partial validation, which is a pain point that `@handrews` had to deal with (pressure from implementers) and what the previous wording of this section tried so hard to allow.

However, if the schema says, "You're getting assertions," then the implementation MUST be able to provide validation or refuse to process the schema.

Reply:
The last scenario is `<format-assertion>: false`.  This means, "If you support asserting these things, that's great, please do.  If not, it's cool; they'll be annotations (because you don't understand them)."

Reply:
In the end, we have

| Annotation<br>Vocab | Assertion<br>Vocab | Impl Assert<br>Support | App Config | Result |
|--|--|--|--|--|
| false/true | n/a | no | n/a | annotation |
| false/true | n/a | yes | annotation | annotation |
| false/true | n/a | yes | assertion | assertion |
| n/a | false | no | n/a | annotation |
| n/a | false | yes | annotation/assertion | assertion |
| n/a | true | no | n/a | fail |
| n/a | true | yes | annotation/assertion | assertion |

Reply:
> But this bit of the spec allows the client/application to say, "I know I'm supposed to get annotations, but this library I'm using supports asserting formats, and that'd save me some work, so I'd like to use that." It gives the choice to the client.

My main point here is that if the implementation can do this, the setting MUST default to NOT doing it. Users must explicitly opt in. Otherwise, there is *no* way of a schema being processed as annotation only (with any combination of $schema and $vocabulary settings) with an implementation that can also handle assertions.

Reply:
> the setting MUST default to NOT doing it

Absolutely.  That is covered on the next line: "... evaluation and MUST be disabled by default."